src/pathfinder/water_regions.cpp

```diff
@@ -409,8 +400,12 @@ void VisitWaterRegionPatchNeighbors(const WaterRegionPatchDesc &water_region_pat
  */
 void AllocateWaterRegions()
 {
+	const int number_of_regions = static_cast<size_t>(GetWaterRegionMapSizeX()) * GetWaterRegionMapSizeY();
```

Comment:
It's not really clear whether you actually want a size_t or an int here?

Reply:
What is clear is that I needed more sleep when I wrote this